Server/src/usercoordinator.erl

```diff
@@ -2,15 +2,26 @@
 -export([init/1]).
 
 %%Author: Tim Jonasson
-%%Version: 2.1
+%%Version: 2.2
+%%collaborator Kosara Golemshinska
 
 %Initializes the usercoordinator
 init(Socket) -> 
   inet:setopts(Socket, [{active, true}]),
+  % Start trapping exits.
+  process_flag(trap_exit, true),
   loop(Socket, []).
 
-  
+ 
 loop(Socket, Diagrams) -> 
+  % If the diagram list has at least 1 item,
+  % get its Pid and link to it. This way, error
+  % propagation is ensured. 
+  if length(Diagrams) > 0 ->
```

Comment:
Is this needed?

Reply:
Yes, if the list is empty, hd/1 will throw an error